templates/terraform/resource.erb

```diff
@@ -322,7 +322,7 @@ func resource<%= resource_name -%>Update(d *schema.ResourceData, meta interface{
 	<% elsif object.custom_code.encoder -%>
 	obj, err = resource<%= resource_name -%>Encoder(d, meta, obj)
 	<% end -%>
-	url, err := replaceVars(d, config, "<%= self_link_url(object) -%>")
+	url, err := replaceVars(d, config, "<%= update_url(object, object.update_url) -%>")
```

Comment:
I think this logic belongs in core, and should be in PCA also.  It does have to get done next quarter if not today.

Reply:
Fair. In the interest of keeping PRs smallish, I'm not going to do it now (doing so would merit adding Autoscaler to the other providers to show that it works, which means adding examples and testing to make sure they actually work. this is non-trivial enough that I'd like to take my time on it)